### PR TITLE
Add validation check icon and padding

### PR DESCRIPTION
**Pull request by devicedev**
https://etvas.atlassian.net/browse/ETV-12290

### Notes
Increased padding right from 40px to 48px for aesthetic purposes

Before:
image
After:
image
image
image
image
image

### CODE REVIEW

File: src/Input/Input.tsx [hunk 1]
```diff
@@ -113,7 +113,7 @@ export const Input = forwardRef<HTMLInputElement, InputProps>(
 
     const icRightHidden =
       !currentIcRight || (currentIcRight === 'check' && !showValidationCheck)
-    const inputPaddingRight = icRightHidden ? 2 : 10
+    const inputPaddingRight = icRightHidden ? 2 : 12
```
Comment (ohnu) [inline src/Input/Input.tsx:116]: is it safe to modify it here for all consumers of this input ?

Comment (devicedev) [reply src/Input/Input.tsx:116]: I think so

File: src/PhoneNumberInput/PhoneNumberInput.tsx [hunk 2]
```diff
@@ -341,10 +344,17 @@ const StyledPhoneNumberWrapper = styled.div<StyledPhoneNumberWrapperProps>(
...
+  paddingRight: number
+}
+
+const StyledPhoneNumberInput = styled.input<StyledPhoneNumberInputProps>(
   css(styles.phoneNumberInput as SystemStyleObject) as any,
   variant({ variants }),
-  `border: none;`
+  `border: none;`,
+  ({ paddingRight }: StyledPhoneNumberInputProps) => ({
+    paddingRight: `${paddingRight * 4}px`
```
Comment (hututudor) [inline src/PhoneNumberInput/PhoneNumberInput.tsx:356]: You shouldn't multiply by 4 directly here ... it is a different behaviour than the normal size multiplication for other props.

Either use the css function from the styled system or find the real problem why this isn't automatically mapped to the theme  in the first place

File: src/PhoneNumberInput/PhoneNumberInput.tsx [hunk 3]
```diff
@@ -266,15 +268,16 @@ const PhoneNumberInput = forwardRef<HTMLInputElement, Props>((props, ref) => {
...
-            <Icon
-              mr={5}
-              size='small'
-              name={currentIcRight}
-              color={currentIcRightColor}
-              spin={currentIcRight === 'loading'}
-            />
-          )}
+          {icStateIsNotIconToggle() &&
+            (currentIcRight !== 'check' || showValidationCheck) && (
```
Comment (hututudor) [inline src/PhoneNumberInput/PhoneNumberInput.tsx:272]: Extract this boolean condition in a constant as it is quite hard to read and understand what it does

File: src/Input/Input.tsx (hunk 1 above)
Comment (hututudor) [reply src/Input/Input.tsx:116]: Maybe check for every input type? Modifying the base component seems very unsafe

File: src/PhoneNumberInput/PhoneNumberInput.tsx (hunk 2 above)
Comment (devicedev) [reply src/PhoneNumberInput/PhoneNumberInput.tsx:356]: I'll look into how we could do this without explicit multiplication

Comment (hututudor) [reply src/PhoneNumberInput/PhoneNumberInput.tsx:356]: > image

Nice, ai mai gasit un loc unde trebuie schimbat

Comment (devicedev) [reply src/PhoneNumberInput/PhoneNumberInput.tsx:356]: 🤡

Comment (devicedev) [reply src/PhoneNumberInput/PhoneNumberInput.tsx]: c5655fcd1277b61721b0c8519eeb6b3adb3ba557

File: src/Input/Input.tsx (hunk 1 above)
Comment (devicedev) [reply src/Input/Input.tsx:121]: From what I saw, every input still renders as expected even with this modification.

Comment (devicedev) [reply src/Input/Input.tsx]: 9227145d1429dd00450d17d8b7abfaec3edf5c51

File: src/PhoneNumberInput/PhoneNumberInput.tsx (hunk 3 above)
Comment (devicedev) [reply src/PhoneNumberInput/PhoneNumberInput.tsx]: 9227145d1429dd00450d17d8b7abfaec3edf5c51

**Review by hututudor — APPROVED** [review]

**Pull request merged**